tekton/koparse/koparse.py

```diff
@@ -62,9 +62,13 @@ def parse_release(base: str, path: str) -> List[str]:
     images = []
     with open(path) as f:
         for line in f:
-            match = re.search(base + ".*" + DIGEST_MARKER + ":[0-9a-f]*", line)
-            if match:
-                images.append(match.group(0))
+            pattern = base + "[0-9a-z\-]+" + DIGEST_MARKER + ":[0-9a-f]*"
+            match = re.search(pattern, line)
+            while match is not None:
+                image = match.group(0)
+                images.append(image)
+                line = re.sub(image, "found", line)
```

Comment:
is it what is expected? to re.sub to `found` ?

Reply:
@chmouel I… need to replace the `found` match so that it doesn't loop forever when *re* matching things… It's not super sexy

Reply:
ah then perhaps something like this would be much prettier : 


```python
import re

YAML = """
      - args: [
        "-logtostderr",
        "-stderrthreshold",
        "INFO",
        "-kubeconfig-writer-image", "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/kubeconfigwriter@sha256:68453f5bb4b76c0eab98964754114d4f79d3a50413872520d8919a6786ea2b35", "-creds-ige", "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/creds-init@sha256:67448da79e4731ab534b91df08da547bc434ab08e41d905858f2244e70290f48",
        "-git-image",
        "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init@sha256:7d5520efa2d55e1346c424797988c541327ee52ef810a840b5c6f278a9de934a",
      ]
"""
DIGEST_MARKER = "@sha256"
base = "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/"
pattern = re.compile(base + r"[0-9a-z\-]+" + DIGEST_MARKER + r":[0-9a-f]*")
for line in YAML.split("\n"):
    findall = re.findall(pattern, line)
    if findall:
        print(findall)

```

Reply:
(but that can be done in a followup)